applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_postprocess.cpp

```diff
@@ -263,8 +299,11 @@ namespace Kratos
                 this->AssembleNodalSensitivityContribution(
                     rOutputVariable, sensitivity_vector[k], r_geom);
             }
-        }
 
+            mSensitivityMatrixI[elem_i.Id()] = sensitivity_matrix[k];
+        }       
+
+        mSolutionStep++;
```

Comment:
Increasing the solution step in that way only works in the presence of one nodal sensitivity variable. I know you are only interested in `SHAPE` but this general framework should also work for other situations.

Reply:
is there a way in Kratos to get the solution step?
If not then I will think about another way for this

Reply:
I fixed this, now it gets the information from ProcessInfo